### ISSUE TITLE
POW output mismatch against CPU reference implementation

### ISSUE THREAD

**Issue by karthikkattaopenxpu** [tenstorrent/tt-metal#23529]
### Describe the bug

When computing tensor power operations (e.g., tensor ** 2), the TT backend produces results that differ from the reference CPU computation. 

### Steps to reproduce the issue

```
test_vals = torch.tensor([9.0], dtype=torch.bfloat16)
expected = torch.tensor([81.0], dtype=torch.bfloat16)  # 9^2
actual = torch.ops.aten.pow.Tensor_Scalar(test_vals.to(device), 2)
```

### Expected behavior
Expected result: 81.0 (bfloat16)
Actual result: 78.0 (bfloat16)

**Comment by fahmedchanTT** [comment]
@ayerofieiev-tt  or @jmalone-tt Can you help out here ?

**Comment by cmaryanTT** [comment]
This is a known issue. It's being worked on.

**Comment by karthikkattaopenxpu** [comment]
Any update on this? This is currently blocking the DeepSeek model completion, as the POW mismatch is affecting the final output.

**Comment by cmaryanTT** [comment]
@karthikkattaopenxpu - we're actively working on it. We have some quick ideas that should improve accuracy.

Can you please provide some info on how this is being called in your model. pow(tensor,tensor) can have a different solution than pow(tensor, scalar)

**Comment by KalaivaniMCW** [comment]
Hi @karthikkattaopenxpu , 
In your model, will the exponents always be integers or they could be fractional values as well ?
I would like to know if the exponent is passed as scalar / tensor to ttnn.pow
if it is passed as single scalar value , you will be getting 81 as the result. 
I would like to know what your use case is. Tensor-Scalar or Tensor-Tensor

<img width="1350" alt="Image" src="https://github.com/user-attachments/assets/f088094f-c8a2-4540-8e34-7054a222333c" />

**Comment by karthikkattaopenxpu** [comment]
Thanks. DeepSeek sends the exponent as a scalar, so we updated our code to pass a scalar instead of a scalar tensor. This is helping to unblock the model for now. We're keeping the issue open to address the mismatch between ScalarTensor and Tensor cases.

**Comment by cmaryanTT** [comment]
@karthikkattaopenxpu - can you provide the exact call from your Deepseek model, with some shape details. Also, is the exponent fixed or variable?

**Comment by karthikkattaopenxpu** [comment]
**Deepseek :** 

OxOp:OxOperation(opcode=powTensor_Scalar, inputs=[Float[1 x 16 x 1536,24576 x 1536 x 1,dev:0,1182],Int64[,,host,-3236]={2}], outputs=[Float[1 x 16 x 1536,24576 x 1536 x 1,dev:0,1183]], execution_flags=)

This case is resolved as per your suggestion.

**Cogvideo :** 

 OxOp:OxOperation(opcode=powScalar, inputs=[Int64[,,host,-2]={10000},BFloat16[1,1,dev:0,1]], outputs=[BFloat16[1,1,dev:0,2]], execution_flags=)  

// This case still has issue. The mismatch between reference and tt output is quite high.  Could you look into this case also ?
p outputs_ref
tensor([15335424.], dtype=torch.bfloat16)
print (outputs_dev)
tensor([12845056.], dtype=torch.bfloat16, device='cuda:0')

**Comment by KalaivaniMCW** [comment]
@karthikkattaopenxpu 

In this case the base/input is a Scalar and the exponent is a Tensor (this will also become a Tensor-Tensor case though) or is it Tensor-Tensor ?

Is it possible to give me the input values that were passed for this specific output [15335424.] , [12845056.] ?
input/base = ___?
exponent = ___?
I want to verify if the drop is indeed propagated from exp or log. 

`OxOp:OxOperation(opcode=powScalar, inputs=[Int64[,,host,-2]={10000},BFloat16[1,1,dev:0,1]], outputs=[BFloat16[1,1,dev:0,2]], execution_flags=)`

**Comment by karthikkattaopenxpu** [comment]
Yeah this will also become Tensor - tensor case which is needed for Cogvideo. I will try the work around and make it as Tensor - Scalar case and see for now. Here are the values you asked : 

input/base = 10000
exponent = 1.7984

```
    exponent = torch.rand([1], dtype=dtype_default, device="cuda")
    exponent[0] = 1.7984
    outputs_ref = None
    if compute_reference:
        outputs_ref = torch.ops.aten.pow.Scalar(10000, exponent)
    exponent = exponent.to(device)
    outputs_dev = torch.ops.aten.pow.Scalar(10000, exponent)
```

**Comment by KalaivaniMCW** [comment]
Thanks for sharing the reference values @karthikkattaopenxpu 
looking at the exponent, the computation will  depend on the exp/log formula. 
Will update after some investigation & tests.

**Comment by fahmedchanTT** [comment]
@nmauriceTT Can this case be closed ? The referenced MR is merged.
